### PR TITLE
Restore printing of archive contents before upload

### PR DESCRIPTION
**Pull request by dschuff**
This should work again with the new bot infrastructure.
Also remove redundant check in subroutine, which makes things
easier to test locally.

### CODE REVIEW

File: src/build.py [hunk 1]
```diff
@@ -293,8 +293,6 @@ def CopyLibraryToSysroot(library):
 def Archive(directory, print_content=False):
   """Create an archive file from directory."""
   # Use the format "native" to the platform
-  if not buildbot.IsBot():
-    return
```
Comment (sbc100) [inline src/build.py]: Did you mean to remove this?

File: src/build.py [hunk 2]
```diff
@@ -1208,7 +1206,7 @@ def ArchiveBinaries():
   # TODO(sergiyb): Restore printing list of binaries on Linux and Mac once it
   # works. See https://crbug.com/916775 and https://crbug.com/940663
```
Comment (sbc100) [inline src/build.py]: Remove the TODO too?

File: src/build.py (hunk 1 above)
Comment (dschuff) [reply src/build.py]: Yes, see the "Also" part of the comment.

File: src/build.py (hunk 2 above)
Comment (dschuff) [reply src/build.py]: Done.

**Pull request merged**